snuba/subscriptions/data.py

```diff
@@ -129,6 +141,57 @@ class LegacySubscriptionData(SubscriptionData):
 
     conditions: Sequence[Condition]
     aggregations: Sequence[Aggregation]
+    organization: Optional[int] = None
+    limit: int = 1
+    offset: int = 0
+    granularity: int = 3600
+
+    @staticmethod
+    def has_crash_rate_alert_aggregation(aggregations: Sequence[Aggregation]) -> bool:
+        for aggregation in aggregations:
+            try:
+                if aggregation[2] and bool(
+                    CRASH_RATE_ALERT_PATTERN.match(aggregation[2])
+                ):
+                    return True
+            except IndexError:
+                continue
+        return False
+
+    @classmethod
+    def update_sessions_subscriptions_with_session_params(
```

Comment:
`subscriptions.data` module should be fairly generic and work for any datasets we want to have subscriptions for in the future. I don't think you should be adding sessions specific methods here as this does not seem to be the right place for them. More concerning to me is the fact that we now seem to be calling these functions on all subscriptions even those that have nothing to do with sessions. Perhaps all of this logic would be better as defined on the dataset/entity or in a query processor instead?

Reply:
That makes sense!